File: node/juggernaut/client/examples/juggernaut_observer.rb
class JuggernautObserver < ActiveRecord::Observer
  observe :activity, :user
  
  def after_create(rec)
    publish(:create, rec)
  end
  
  def after_update(rec)
    publish(:update, rec)
  end
  
  def after_destroy(rec)
    publish(:destroy, rec)
  end
  
  protected
    def publish(type, rec)
      Juggernaut.publish(
        Array(rec.sync_clients).map {|c| "/sync/#{c}" }, 
        {:type => type, :id => rec.id, 
         :klass => rec.class.name, :record => rec}
      )
    end
end

File: node/juggernaut/client/examples/juggernaut_observer.js
// Assumes you're using SuperModel
// http://github.com/maccman/supermodel-js

jQuery(function($){
  var jug = new Juggernaut;
  jug.subscribe("/sync/your_user_id", function(sync){
    var klass = eval(sync.klass);
    switch(sync.type) {
      case "create":
        klass.create(sync.record);
        break;
      case "update":
        klass.update(sync.id, sync.record);
        break;
      case "destroy":
        klass.destroy(sync.id);
        break;
      default:
        throw("Unknown type:" + type);
    }
  });
})